hathor/p2p/netfilter/matches.py

```diff
@@ -27,6 +27,18 @@ class NetfilterMatch(ABC):
     def match(self, context: 'NetfilterContext') -> bool:
         raise NotImplementedError
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, NetfilterMatch):
+            return NotImplemented
+        eq_type = type(self) == type(other)
+        eq_attrs = True
+        for key in self.__dict__.keys():
```

Comment:
I feel this kind of comparison creates bugs that are hard to find. For instance, it wouldn't work for `NetfilterMatchRemoteURL` since it has many internal attributes that will differ.

Why don't you implement a mixin for both "and" and "or" operation. Something like this:

```
class TwoVariablesMixin:
    def __eq__(self, other):
        if notinstance(other, NetfilterMatch):
            return NotImplemented
        if type(self) != type(other):
            return False
        return self.a == other.a and self.b == other.b
```

And you'd also need to implement custom `__eq__` for `NetfilterMatchPeerId` and `NetfilterMatchRemoteURL`.

Reply:
This code does not exist anymore